packages/nextjs/src/utils/withSentry.ts

```diff
@@ -74,7 +74,57 @@ export const withSentry = (origHandler: NextApiHandler): WrappedNextApiHandler =
       }
 
       try {
-        return await origHandler(req, res);
+        const handlerResult = await origHandler(req, res);
+
+        // Temporarily mark the response as finished, as a hack to get nextjs to not complain that we're coming back
+        // from the handler successfully without `res.end()` having completed its work.  This is necessary (and we know
+        // we can do it safely) for a few reasons:
+        //
+        // - Normally, `res.end()` is sync and completes before the request handler returns, as part of the handler
+        //   sending data back to the client. As soon as the handler is done, nextjs checks to make sure that the
+        //   response is indeed finished. (`res.end()` signals this by setting `res.finished` to `true`.) If it isn't,
+        //   nextjs complains.("Warning: API resolved without sending a response for <url>.")
+        //
+        // - In order to prevent the lambda running the route handler from shutting down before we can send events to
+        //   Sentry, we monkeypatch `res.end()` so that we can call `flush()`, wait for it to finish, and only then
+        //   allow the response to be marked complete. This turns the normally-sync `res.end()` into an async function,
+        //   which isn't awaited because it's assumed to still be sync. As a result, nextjs runs the aforementioned
+        //   check before the now-async `res.end()` has had a chance to set `res.finished = false`, and therefore thinks
```

Comment:
When writing docs, I think using "booleanistic" expressions makes the comments easier to understand. We can also use words if none of the options convince you: "had a chance to set `res.finished` to false".
```suggestion
        //   check before the now-async `res.end()` has had a chance to set `res.finished == false`, and therefore thinks
```

Reply:
It's a purposeful choice, though, based on the meaning: I use "booleanistic" syntax when I'm taking about testing a value ("if `x == 5`, that means that...") but "assignmentistic" (LOL) syntax when I'm talking about setting a value (as above).